<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SpriteEditor</class>
 <widget class="QMainWindow" name="SpriteEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>993</width>
    <height>644</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sprite Editor</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="AobaGraphicsView" name="frameEditor"/>
    </item>
   </layout>
  </widget>
  <widget class="QDockWidget" name="dockWidget_3">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Preview</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QWidget" name="widget_2" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="animationRestartButton">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../resources/Resources.qrc">
            <normaloff>:/bright/track-start</normaloff>:/bright/track-start</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="animationPlayButton">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../resources/Resources.qrc">
            <normaloff>:/bright/play</normaloff>:/bright/play</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="AobaGraphicsView" name="preview"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_4">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Image palette</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QWidget" name="widget_3" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="addImagesButton">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../resources/Resources.qrc">
            <normaloff>:/bright/plus</normaloff>:/bright/plus</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="removeImagesButton">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../resources/Resources.qrc">
            <normaloff>:/bright/cross</normaloff>:/bright/cross</iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>120</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QComboBox" name="imagesFilter"/>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="imageList">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::NoDragDrop</enum>
       </property>
       <property name="defaultDropAction">
        <enum>Qt::IgnoreAction</enum>
       </property>
       <property name="iconSize">
        <size>
         <width>48</width>
         <height>48</height>
        </size>
       </property>
       <property name="movement">
        <enum>QListView::Static</enum>
       </property>
       <property name="resizeMode">
        <enum>QListView::Adjust</enum>
       </property>
       <property name="viewMode">
        <enum>QListView::IconMode</enum>
       </property>
       <property name="uniformItemSizes">
        <bool>false</bool>
       </property>
       <property name="selectionRectVisible">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_6">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Timeline</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_6">
    <layout class="QVBoxLayout" name="verticalLayout_10">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="AobaTimeline" name="timeline" native="true">
       <property name="widgetResizable" stdset="0">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="spriteToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>true</bool>
   </property>
   <property name="floatable">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QToolBar" name="animationToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionCreateCustomAnimation"/>
  </widget>
  <widget class="QToolBar" name="frameBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionFlipX"/>
   <addaction name="actionFlipY"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Frame properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QScrollArea" name="frameScrollArea">
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>257</width>
          <height>159</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QWidget" name="widget" native="true">
           <layout class="QFormLayout" name="formLayout">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Frame Type</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="frameTypeCombo"/>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="frame">
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>134</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="imageToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPriorityImage"/>
   <addaction name="actionStaticImage"/>
   <addaction name="actionSharedImage"/>
   <addaction name="actionImageToForeground"/>
   <addaction name="actionImageToBackground"/>
  </widget>
  <action name="actionFlipX">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/mirrorx</normaloff>:/bright/mirrorx</iconset>
   </property>
   <property name="text">
    <string>Flip X</string>
   </property>
  </action>
  <action name="actionFlipY">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/mirrory</normaloff>:/bright/mirrory</iconset>
   </property>
   <property name="text">
    <string>Flip Y</string>
   </property>
  </action>
  <action name="actionPriorityImage">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/image-priority</normaloff>:/bright/image-priority</iconset>
   </property>
   <property name="text">
    <string>Priority image</string>
   </property>
  </action>
  <action name="actionImageToForeground">
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/sort-ascending</normaloff>:/bright/sort-ascending</iconset>
   </property>
   <property name="text">
    <string>Image to foreground</string>
   </property>
  </action>
  <action name="actionImageToBackground">
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/sort-descending</normaloff>:/bright/sort-descending</iconset>
   </property>
   <property name="text">
    <string>Image to background</string>
   </property>
  </action>
  <action name="actionStaticImage">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/padlock</normaloff>:/bright/padlock</iconset>
   </property>
   <property name="text">
    <string>Static Image</string>
   </property>
  </action>
  <action name="actionSharedImage">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/image-shared</normaloff>:/bright/image-shared</iconset>
   </property>
   <property name="text">
    <string>Shared Image</string>
   </property>
  </action>
  <action name="actionCreateCustomAnimation">
   <property name="icon">
    <iconset resource="../resources/Resources.qrc">
     <normaloff>:/bright/plus</normaloff>:/bright/plus</iconset>
   </property>
   <property name="text">
    <string>Create custom animation</string>
   </property>
   <property name="toolTip">
    <string>Create a custom animation</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>AobaTimeline</class>
   <extends>QWidget</extends>
   <header location="global">maker/common/AobaTimeline.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AobaGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>maker/common/AobaGraphicsView.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/Resources.qrc"/>
 </resources>
 <connections/>
</ui>
